coreos-installer

```diff
@@ -664,31 +682,87 @@ log() {
     echo "$1" >&2
 }
 
+#########################################################
+#Update bootloader if needed
+#########################################################
+write_bootloader() {
+    if [[ "$arch" == "s390x" ]]; then
+        log "Updating zipl"
+        mkdir -p /mnt/boot_partition
+        mount_boot_partition /mnt/boot_partition
+        trap 'umount /mnt/boot_partition; trap - RETURN' RETURN
+
+        blsfile=$(ls /mnt/boot_partition/loader/entries/*.conf)
+        echo "$(grep options $blsfile | cut -d' ' -f2-) $(cat /tmp/s390x_opts) ignition.firstboot rd.neednet=1 ip=dhcp" > /tmp/zipl_prm
+        zipl --verbose -p /tmp/zipl_prm -i $(ls /mnt/boot_partition/ostree/*/*vmlinuz*) -r $(ls /mnt/boot_partition/ostree/*/*initramfs*) --target /mnt/boot_partition
+        if [[ $? -ne 0 ]]; then
+            log "failed updating bootloder"
+            exit 1
+        fi
+        # this allows a reboot into new installed disk
+        chreipl "${DEST_DEV}"
+    fi
+}
+#########################################################
+#Handle s390x zFCP SCSI devices
+#########################################################
+write_image_to_zfcp_disk() {
+    log "Extracting disk image"
+    mount_tmpfs $(( $($DECOMPRESSION_TOOL ${imagefile_compressed} | wc --bytes) / 1024 / 1024 + ${TMPFS_MBSIZE} ))
```

Comment:
Probably want to add some additional padding.

Reply:
Trying to get a proper number for padding ...

Reply:
@arithx : Just in case it's not clear: At this line, tmpfs at `/mnt/dl` is already mounted, and compressed image was downloaded and stored there. We need to extract the image to read its partition table, thus requiring to remount `/mnt/dl` with larger size.

Reply:
I cannot come up with a good reasoning for a number, so I added `100` for padding ...

Reply:
Feel free to resolve or add more comments, many thanks.